docs-chef-io/content/server/upgrade_server_ha_v2.md

```diff
@@ -166,10 +166,74 @@ Upgrading from Chef Backend 1.x to Chef Backend 2.x requires full cluster downti
 | Running Version | Upgrade To Version | Upgrade Type |
 |---------|---------|------|
 | 2.x | 2.x | Rolling |
-| 2.1 | 3.0 | Rolling |
+| 2.1 | 3.0 | Rolling if the created string in curl -XGET localhost:9200/chef/_settings/index.version.created*?pretty\&human is 5.5 or greater|
+| 2.1 | 3.0 | Downtime if the created string in curl -XGET localhost:9200/chef/_settings/index.version.created*?pretty\&human is 5.4 or lesser|
 | 2.0 | 3.0 | Downtime |
 
-Upgrading from Chef Backend 2.0 to directly Chef Backend 3.0 requires full cluster downtime and is not supported.
+Chef Backend clusters that were originally installed with a version of Chef Backend before 2.1.0 will need to delete and reindex elasticsearch indices after upgrading.
+
+To determine if your indexes need to be replaced run the command:
+
+```bash
+curl -XGET localhost:9200/chef/_settings/index.version.created*?pretty\&human
+```
+
+and look for the version number in "created_string"
+
+If "created_string" : "2.3.1" or any version less than "5.6.16" you will need to perform these additional steps:
+
+1. Run this command on all front end nodes to stop them:
+   ```bash
+   sudo chef-server-ctl stop
+   ```
+
+2. Disable all frontends other than one
+
+   On frontends running Infra Server 14.6.32 or later you can disable the API temporarily from remote access.
+   ```bash
+   sudo chef-server-ctl maintenance on
```

Comment:
If you stop the front end in the prior step with `chef-server-ctl stop` you can't turn on maintenance mode because redis will be down.

```
/opt/opscode/embedded/lib/ruby/gems/2.7.0/gems/redis-4.5.1/lib/redis/client.rb:398:in `rescue in establish_connection': Error connecting to Redis on 127.0.0.1:16379 (Errno::ECONNREFUSED) (Redis::CannotConnectError)
```

Reply:
The steps probably are:
1. stop all the front ends
2. do the backend stuff
3. chef-server-ctl reconfigure on one of the frontends - works if it is stopped, leaves it running
4. sudo chef-server-ctl maintenance on # so nodes don't search while you don't have an index
5. reindex commands 
6. knife test stuff # works if run from the server that is in maintenance mode.
7.    sudo chef-server-ctl maintenance off
8. start the rest of the front ends

Reply:
@PrajaktaPurohit Let me know which is the right approach

Reply:
I would follow @btm 's sequencing overall in the doc

Reply:
There are some minor tweaks that we will be doing wrt having the internal index version to be at the latest and not having to start services on FE that are either not out of the lb or in maint mode. But other that that those are the broad steps that we would be following.